bot/utils/time.py

```diff
@@ -326,3 +330,14 @@ def until_expiration(expiry: Optional[Timestamp]) -> str:
         return "Expired"
 
     return format_relative(expiry)
+
+
+def round_datetime(dt: datetime.datetime) -> datetime.datetime:
+    """
+    Round a datetime object to the nearest second.
+
+    Resulting datetime objects will have microsecond values of 0, useful for delta comparisons.
+    """
+    if dt.microsecond >= 500000:
+        dt += datetime.timedelta(seconds=1)
+    return dt.replace(microsecond=0)
```

Comment:
Can you give an example case where the representation is improved with this function? Most deltas are in the hours or days so I'm not sure what called for this addition.

Reply:
So that was the rationale for this change, since the minimum accuracy of our `humanize_delta` is 1 second, the rounding will trim off microseconds via rounding.

The original issue was that the `humanize_delta` call matches time units until the last unit is exhausted, which, in the case of infractions, due to code and network latency, the delta between `inserted_at` (calculated at entry insertion server side) and `expires_at` (calculated at original command invocation) will *at least* be some microseconds off. In some extreme cases, this will be up to 10s or more.

In most cases, the entries are within 1 second in delta, but this is still enough to cause the `59 minutes 59 seconds` display for the original `1h` infraction, for example. This rounding will enable most legacy entries without a historical `last_applied` entry (currently defaults to their original `inserted_at` value at the site side) to still display a most likely whole unit of time.